dags/prio/processor.py

```diff
@@ -0,0 +1,330 @@
+from os import environ
+
+from airflow import DAG
+from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+from airflow.contrib.operators.gcs_to_gcs import (
+    GoogleCloudStorageToGoogleCloudStorageOperator,
+)
+from airflow.operators import PythonOperator
+from airflow.operators.subdag_operator import SubDagOperator
+from prio import dataproc, kubernetes
+
+DEFAULT_ARGS = {
+    "owner": "amiyaguchi@mozilla.com",
+    "depends_on_past": True,
+    "wait_for_downstream": True,
+    "start_date": datetime(2019, 8, 22),
+    "email": [
+        "amiyaguchi@mozilla.com",
+        "hwoo@mozilla.com",
+        "dataops+alerts@mozilla.com",
+    ],
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 0,
+    "retry_delay": timedelta(minutes=15),
+    "schedule_interval": "@daily",
+    "dagrun_timeout": timedelta(hours=2),
+}
+
+# use a less than desirable method of generating the service account name
+IS_DEV = environ.get("DEPLOY_ENVIRONMENT") == "dev"
+ENVIRONMENT = "dev" if IS_DEV else "prod"
+
+PROJECT_ADMIN = GoogleCloudStorageHook("google_cloud_prio_admin").project_id
+PROJECT_A = GoogleCloudStorageHook("google_cloud_prio_a").project_id
+PROJECT_B = GoogleCloudStorageHook("google_cloud_prio_b").project_id
+
+SERVICE_ACCOUNT_ADMIN = "prio-admin-runner@{}.iam.gserviceaccount.com".format(
+    PROJECT_ADMIN
+)
+SERVICE_ACCOUNT_A = "prio-runner-{}-a@{}.iam.gserviceaccount.com".format(
+    ENVIRONMENT, PROJECT_A
+)
+SERVICE_ACCOUNT_B = "prio-runner-{}-b@{}.iam.gserviceaccount.com".format(
+    ENVIRONMENT, PROJECT_B
+)
+
+BUCKET_PRIVATE_A = "moz-fx-prio-{}-a-private".format(ENVIRONMENT)
+BUCKET_PRIVATE_B = "moz-fx-prio-{}-b-private".format(ENVIRONMENT)
+BUCKET_SHARED_A = "moz-fx-prio-{}-a-shared".format(ENVIRONMENT)
+BUCKET_SHARED_B = "moz-fx-prio-{}-b-shared".format(ENVIRONMENT)
+BUCKET_DATA_ADMIN = "moz-fx-data-{}-prio-data".format(ENVIRONMENT)
+BUCKET_BOOTSTRAP_ADMIN = "moz-fx-data-{}-prio-bootstrap".format(ENVIRONMENT)
+
+dag = DAG(dag_id="prio_processor", default_args=DEFAULT_ARGS)
+
+
+# Copy the dependencies necessary for running the `prio-processor staging` job
+# into the relevant GCS locations. This includes the runner and the egg
+# containing the bundled dependencies.
+
+prio_staging_bootstrap = SubDagOperator(
+    subdag=kubernetes.container_subdag(
+        parent_dag_name=dag.dag_id,
+        child_dag_name="bootstrap",
+        default_args=DEFAULT_ARGS,
+        server_id="admin",
+        gcp_conn_id="google_cloud_prio_admin",
+        service_account=SERVICE_ACCOUNT_ADMIN,
+        arguments=[
+            "bash",
+            "-c",
+            "cd processor; prio-processor bootstrap --output gs://{}".format(
+                BUCKET_BOOTSTRAP_ADMIN
+            ),
+        ],
+        env_vars={},
+    ),
+    task_id="bootstrap",
+    dag=dag,
+)
+
+# Run the PySpark job for range-partioning Prio pings.
+prio_staging = SubDagOperator(
+    subdag=dataproc.spark_subdag(
```

Comment:
I think this could use the moz_pyspark_runner if if was modified to take a few addl arguments

Reply:
I'm going to leave this as is, but it might be worthwhile in the future to refactor to increase code reuse.